enterprise/reporting/mediator.go

```diff
@@ -23,6 +23,7 @@ type Mediator struct {
 	ctx       context.Context
 	cancel    context.CancelFunc
 	reporters []types.Reporting
+	stats     stats.Stats
 }
 
 func NewReportingMediator(ctx context.Context, log logger.Logger, enterpriseToken string, backendConfig backendconfig.BackendConfig) *Mediator {
```

Comment:
We can probably pass stats as a dependency to the Mediator from the [setup](https://github.com/rudderlabs/rudder-server/blob/b80d27394b42116f3b0f7e86c71d1ed126230ae4/enterprise/reporting/setup.go#L21) itself.